tools/git/hooks/src/pre_commit.py

```diff
@@ -5,25 +5,45 @@
 import subprocess
 import sys
 import tempfile
+import argparse
 
 def main(): # pragma: no cover
-    files = sys.argv[1:]
-    failed_files = commit_is_ready(files)
-    if failed_files:
-        print("\nYou may run the following command to repeat the check:")
-        print("Hint: You may need to be at the repository's root directory.")
-        print("\n" + " \\\n".join([sys.argv[0]] + failed_files) + "\n")
-        print('Aborting Commit.')
-        sys.exit(1)
+    args = parse_args()
+
+    if args['uncrustify_all_files']:
+        file_names = get_all_files()
+        files_to_check = filter_checkable_files(file_names)
+        check_uncrustify(files_to_check, uncrustify=True)
+    elif args['uncrustify']:
+        file_names = get_modified_files()
+        files_to_check = filter_checkable_files(file_names)
+        check_uncrustify(files_to_check, uncrustify=True)
+    else:
+        failed_files = commit_is_ready()
+        if failed_files:
+            print("\nYou may run the following command to repeat the check: python tools/git/hooks/src/pre_commit.py")
+            print("\nYou may run the following command to uncrustify modified files: python tools/git/hooks/src/pre_commit.py --uncrustify")
+            print("\nYou may run the following command to uncrustify all files: python tools/git/hooks/src/pre_commit.py --uncrustify-all-files")
+            print("Hint: You may need to be at the repository's root directory.")
+            print('Aborting Commit.')
+            sys.exit(1)
     sys.exit(0)
 
 
-def commit_is_ready(file_names=""):
-    """Return False if not ready.  Return True if commit is ready"""
-    if not file_names:
-        file_names = get_modified_files()
+def parse_args():
+    parser = argparse.ArgumentParser(description='pre-commit checks')
 
-    files_to_check = [f for f in file_names if file_exists(f) and file_is_checkable(f)]
+    parser.add_argument('--uncrustify', default = False, action='store_true', help='Uncrustify modified files.')
+    parser.add_argument('--uncrustify-all-files', default = False, action='store_true', help='Uncrustify all files.')
```

Comment:
The commandline interface used to take a list of files, without it the PR check will fail. You probably need to add a positional arugment like this,
```python
parser.add_argument('files', metavar='file', nargs='*', help='list of files to check.')
```
reference:
- https://docs.python.org/3/library/argparse.html#name-or-flags
- https://docs.python.org/3/library/argparse.html#nargs

Reply:
Thank you for catching this. The PR check does not have the list of staged files and therefore needs to pass the list of files on the command line. I have added this option.